apps/vs-code-designer/src/app/utils/dotnet/dotNetVersion.ts

```diff
@@ -0,0 +1,150 @@
+  if (data) {
+    const majorVersion: string | undefined = tryGetMajorVersion(data);
+    if (majorVersion) {
+      return Object.values(DotnetVersion).find((version) => version === majorVersion);
+    }
+  }
+
+  return undefined;
+}
+
+/**
+ * Gets major version of package.
+ * @param {string} data - Package version.
+ * @returns {string | undefined} Major version.
+ */
+export function tryGetMajorVersion(data: string): string | undefined {
+  const match: RegExpMatchArray | null = data.match(/^[~v]?([0-9]+)/i);
+  return match ? match[1] : undefined;
+}
+
+/**
+ * Gets default .Net SDK  version either from open workspace, local cli or backup.
+ * @param {string} context - Command context.
+ * @returns {Promise<DotnetVersion>} Major version.
+ */
+export async function getDefaultDotNetSDKVersion(context: IActionContext): Promise<DotnetVersion> {
+  let version: DotnetVersion | undefined = tryParseDotNetVersion(getWorkspaceSettingFromAnyFolder(dotNetSDKVersionSetting));
+  context.telemetry.properties.runtimeSource = 'VSCodeSetting';
+
+  if (isNullOrUndefined(version)) {
+    version = await tryGetLocalDotNetVersion();
+    context.telemetry.properties.runtimeSource = 'LocalDotNetCli';
+  }
+
+  if (isNullOrUndefined(version)) {
+    version = DotnetVersion.netsdk6;
+    context.telemetry.properties.runtimeSource = 'Backup';
+  }
+
+  return version;
+}
+
+/**
+ * Gets .NET SDK version from local cli command.
+ * @returns {Promise<string | undefined>} .NET SDK version.
+ */
+export async function tryGetLocalDotNetVersion(): Promise<DotnetVersion | undefined> {
+  try {
+    const version: string | null = await getLocalDotNetSDKVersion();
+    if (version) {
+      return tryParseDotNetVersion(version);
+    }
+  } catch (err) {
+    // swallow errors and return undefined
+  }
+
+  return undefined;
+}
+
+/**
+ * Executes version command and gets it from cli.
+ * @returns {Promise<string | null>} .NET SDK version.
+ */
+export async function getLocalDotNetSDKVersion(): Promise<string | null> {
+  const output: string = await executeCommand(undefined, undefined, ext.dotNetCliPath, '--list-sdks');
+
+  // Creates an array of versions from output
+  const sdkList: string[] = output.split(/\r?\n/);
+  const directoryRegex = /\[.*?\]/;
+  let version: string | null = '-1';
+  sdkList.forEach((sdkVersion) => {
+    // Removes directory using regex
+    const match = sdkVersion.match(directoryRegex);
+    if (match) {
+      const directory = match[0];
+      const checkVersion = semver.clean(sdkVersion.replace(directory, ''));
+
+      if (semver.major(checkVersion) == dotNetSDKMajorVersion) {
+        // List of sdk is in-order and therefore the version will be latest.
+        version = checkVersion;
+      }
+    }
+  });
+
+  if (version && version != '-1') {
+    return version;
+  }
+  return null;
+}
+
+export async function getNewestDotNetSDKVersion(
```

Comment:
sure there is a package already available for this somewhere that we can reuse?

Reply:
was not able to find a specific package that parses .NET SDK versions